x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/document_fields/field_parameters/max_shingle_size_parameter.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { getFieldConfig } from '../../../lib';
+import { EditFieldFormRow } from '../fields/edit_field';
+import { UseField, Field } from '../../../shared_imports';
+
+interface Props {
+  defaultToggleValue: boolean;
+}
+
+export const MaxShingleSizeParameter = ({ defaultToggleValue }: Props) => (
+  <EditFieldFormRow
+    title={i18n.translate('xpack.idxMgmt.mappingsEditor.maxShingleSizeFieldTitle', {
+      defaultMessage: 'Set max shingle size',
+    })}
+    description={i18n.translate('xpack.idxMgmt.mappingsEditor.maxShingleSizeFieldDescription', {
+      defaultMessage:
+        'The largest shingle size to index the input with and create subfields for, creating one subfield for each shingle size between 2 and the max value.',
```

Comment:
I had a tricky time parsing this. What do you think of this copy Gail and I came up with?

> The default is three shingle subfields. More subfields enables more specific queries, but increases index size.

Reply:
I think it is great to strive for improving the UI help texts, but we should align them with our docs. So one is not "better" (easier to understand) than the other. This is the current text for the parameter

<img width="977" alt="Screen Shot 2020-01-20 at 12 18 55" src="https://user-images.githubusercontent.com/2854616/72704617-515a9e00-3b7f-11ea-8597-2347ece91048.png">

Also, I am not sure we should add the default (hardcoded) value in the translation. That would be one more place to look for if it ever changes. If we think it adds value, then better to write it in parenthesis (`Default: 3`) and read the value from the parameter definition constant.

Reply:
I love the idea of updating our docs. What do you propose? Would you like to incorporate the change we suggested, and we'll create an issue to update the docs accordingly?

I hear you in terms of keeping the text maintainable. We have hardcoded values in our help text in various places, so this isn't the only place we'll have this problem. How about we hardcode it for now, and create an issue for universally implementing a more robust solution like the one you suggest?

Reply:
> I love the idea of updating our docs. What do you propose? Would you like to incorporate the change we suggested, and we'll create an issue to update the docs accordingly?

Yes let's do that 👍 

Regarding the default value, I haven't looked in details where/when we started using default values in text description but I think we should try as much as possible to look how we did in previous places/forms and repeat the pattern (unless we decided that was a wrong idea).

This is how we did with follower indices and I think it was a great call

<img width="730" alt="Screen Shot 2020-01-21 at 10 57 15" src="https://user-images.githubusercontent.com/2854616/72777979-73b2f100-3c3d-11ea-9491-c879e3204a93.png">

---- 

After we change the default:

<img width="729" alt="Screen Shot 2020-01-21 at 10 57 35" src="https://user-images.githubusercontent.com/2854616/72778001-80374980-3c3d-11ea-8846-712854f7fc86.png">